web/modules/custom/openideal_challenge/src/Service/OpenidealChallengeServiceInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Drupal\openideal_challenge\Service;
+
+/**
+ * OpenidealChallengeServiceInterface file.
+ */
+interface OpenidealChallengeServiceInterface {
+
+  /**
+   * Processing for closing / opening scheduled nodes.
+   *
+   * @param string $operation
+   *   The name of operation 'open/close'.
+   *
+   * @return bool
+   *   TRUE if any node has been closed / opened, FALSE otherwise.
+   */
+  public function challengeOpenCloseOperation($operation = 'open');
```

Comment:
I would like to have two **public** methods: open() and close() that will execute **protected** method challengeOpenCloseOperation() with the appropriate parameter.

Reply:
I've changed the code according to your notes, thanks.